### PR TITLE
GitLab Migration: Normalize Labels Color

### LINKED ISSUE #12779: gitlab migration: fals token error is not reported
gitea v:  1.13.0+dev-583-g090024db3

`Migrate repository from https://gitlab.com/gitea/test_repo failed: error: GitlabDownloader is nil`

### PR DESCRIPTION
**Pull request by 6543**
close #12779

### CODE REVIEW

File: modules/migrations/gitlab.go [hunk 1]
```diff
@@ -215,6 +215,19 @@ func (g *GitlabDownloader) GetMilestones() ([]*base.Milestone, error) {
 	return milestones, nil
 }
 
+func (g *GitlabDownloader) normalizeColor(val string) string {
+	val = strings.TrimLeft(val, "#")
+	val = strings.ToLower(val)
+	if len(val) == 3 {
+		c := []rune(val)
+		val = fmt.Sprintf("%c%c%c%c%c%c", c[0], c[0], c[1], c[1], c[2], c[2])
```
Comment (6543) [inline modules/migrations/gitlab.go:223]: if somebody know a better func to convert `abc` -> `aabbcc` pleace tell me :)

**Comment by 6543** [comment]
to test this case use, just migrate labels of `https://gitlab.com/fdroid/fdroid-website`

File: modules/migrations/gitlab.go (hunk 1 above)
Comment (zeripath) [reply modules/migrations/gitlab.go:223]:
```go
val := "abc"
doubleVal := make([]rune,0, len(val)*2)
for _, v := range val {
	doubleVal = append(doubleVal, v, v)
}
val = string(doubleVal)
```

Comment (zeripath) [reply modules/migrations/gitlab.go:223]: but it's not really better - although it's more efficient

**Review by zeripath — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
🚀

**Pull request merged**